src/controllers/hid/hidiothread.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <QAtomicInteger>
+#include <QThread>
+#include <map>
+
+#include "controllers/controller.h"
+#include "controllers/hid/hiddevice.h"
+#include "controllers/hid/hidioreport.h"
+#include "util/compatibility/qmutex.h"
+#include "util/duration.h"
+
+enum class HidIoThreadState {
+    Initialized,
+    OutputActive,
+    InputOutputActive,
+    StopWhenAllReportsSent,
+    StopRequested,
+    Stopped,
+};
+
+class HidIoThread : public QThread {
+    Q_OBJECT
+  public:
+    HidIoThread(hid_device* pDevice,
+            const mixxx::hid::DeviceInfo& deviceInfo);
+    ~HidIoThread() override;
+
+    void run() override;
+
+    /// Sets the state of the HidIoThread lifecycle.
+    /// Returns immediately with true, if the compared thread state, was equal to the expected state within the timeout.
+    /// Returns immediately with false, if the compared thread state, wasn't equal to the expected state and no timeout is specified
+    /// Returns with false (Debug assert in developer mode), if timeout is reached
+    bool testAndSetThreadState(HidIoThreadState expectedState,
```

Comment:
- Please split into dedicated functions as proposed
- These kind of functions are candidates for `[[nodiscard]]`